### PR TITLE
CDS-57577-CDS-54890

### PR DESCRIPTION
**Pull request by roshnisarangadharan**
Added info about Bitbucket connector and GitOpsSync step

# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by meenaravichandran1 — APPROVED** [review]

File: docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md [hunk 1]
```diff
@@ -237,13 +237,32 @@ A Sync brings the live state to its desired state by applying the declarative de
 
 ![](./static/harness-git-ops-application-set-tutorial-40.png)
 
-When you click **Sync**, Harness will use the ApplicationSet to create the new Harness Applications for the dev and prod clusters.
+<details>
+<summary>SYNC button</summary>
 
-1. In the GitOps Application, click **SYNC**.
+When you select **Sync**, Harness will use the ApplicationSet to create the new Harness Applications for the dev and prod clusters.
```
Comment (StrawberryPad) [inline docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md:243]:
```suggestion
When you select **Sync**, Harness uses the ApplicationSet to create the new Harness Applications for the dev and prod clusters.
```

File: docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md [hunk 2]
```diff
@@ -237,13 +237,32 @@ A Sync brings the live state to its desired state by applying the declarative de
...
+
+1. In the GitOps Application, select **SYNC**.
    
    ![](./static/harness-git-ops-application-set-tutorial-41.png)
 
-2. In the Sync settings, click **Synchronize**. Synchronization will take a minute.
+2. In the Sync settings, select **Synchronize**. Synchronization will take a minute.
```
Comment (StrawberryPad) [inline docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md:249]: I'm not sure what the image above this is showing me. I also don't see Sync settings in the image, so this step is a bit confusing. Maybe in Step 1 you could give a bit more context or explain what the image is... like 'the xx page appears' or something before the image.

File: docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md [hunk 3]
```diff
@@ -237,13 +237,32 @@ A Sync brings the live state to its desired state by applying the declarative de
 
 ![](./static/harness-git-ops-application-set-tutorial-40.png)
 
-When you click **Sync**, Harness will use the ApplicationSet to create the new Harness Applications for the dev and prod clusters.
+<details>
+<summary>SYNC button</summary>
```
Comment (StrawberryPad) [inline docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md:241]: If this is a procedure the user should do or can do, I don't think it should be hidden in an expandable block.

File: docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md [hunk 4]
```diff
@@ -237,13 +237,32 @@ A Sync brings the live state to its desired state by applying the declarative de
...
    
    ![](./static/harness-git-ops-application-set-tutorial-41.png)
 
-2. In the Sync settings, click **Synchronize**. Synchronization will take a minute.
+2. In the Sync settings, select **Synchronize**. Synchronization will take a minute.
+
+</details>
+
+<details>
+<summary>GitOpsSync execution step</summary>
```
Comment (StrawberryPad) [inline docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md:254]: Same comment. Is this a procedure the user can/should do? If so, it shouldn't be in an expandable.

File: docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md [hunk 5]
```diff
@@ -352,13 +371,20 @@ The Harness Service represents what you're deploying. In this case, we're deploy
 1. In **Select Service**, click **New Service**.
 2. In **Name**, enter **PR Example**.
 3. In **Manifests**, click **Add Release Repo Manifest**.
-4. In **Release Repo Store**, select **Github**.  
+4. In **Release Repo Store**, select one of the following repositories.  
 
-Now we'll add a Github Connector to tell Harness where to pull the config.json from.
+```mdx-code-block
```
Comment (StrawberryPad) [inline docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md:376]: Can the tabs code block be indented?

File: docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md [hunk 6]
```diff
@@ -352,13 +371,20 @@ The Harness Service represents what you're deploying. In this case, we're deploy
...
+<Tabs>
+    <TabItem value="GitHub" label="GitHub" default>
+```
 
-#### Add Github Connector
+Now we'll add a Github Connector to tell Harness where to pull the config.json from.
 
-1. In **Github Connector**, click **New Github Connector**.
+1. In **Github Connector**, select **New Github Connector**.
 2. Enter the following Github Connector settings:
```
Comment (StrawberryPad) [inline docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md:388]: The settings shouldn't be numbered steps. More of a bullet list.

Comment (StrawberryPad) [reply docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md:388]: (I know that's beyond what you're doing. :-) But it's glaring.

File: docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md [hunk 7]
```diff
@@ -372,10 +398,39 @@ Now we'll add a Github Connector to tell Harness where to pull the config.json f
    You will store the PAT in a [Harness Text Secret](/docs/platform/Secrets/add-use-text-secrets). For details on Secrets Management, go to [Harness Secrets Management Overview](/docs/platform/Secrets/Secrets-Management/harness-secret-manager-overview).
```
Comment (StrawberryPad) [inline docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md:398]: Please indent lines 394 (image) and 396 ("You will store the PAT...")

**Review by StrawberryPad — CHANGES_REQUESTED** [review]
I was a bit confused by the expandable sections, and made some other style/formatting comments.

File: docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md (hunk 6 above)
Comment (roshnisarangadharan) [reply docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md:388]: @StrawberryPad It is a procedure, shouldn't it be numbered?

File: docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md [hunk 8]
```diff
@@ -372,10 +357,39 @@ Now we'll add a Github Connector to tell Harness where to pull the config.json f
...
+</TabItem>
+<TabItem value="Bitbucket" label="Bitbucket">
+```
+1. In **Bitbucket Connector**, select **New Bitbucket Connector**.
+2. Enter the following Bitbucket Connector settings:
+    1. **Name:** enter **gitops-bitbucket-cloud**.
+    2. **URL Type:** select **Repository**.
+    3. **Connection Type:** select **HTTP**.
+    4. **GitHub Repository URL:** enter the HTTP URL for repo you used for your ApplicationSet, such as `https://bitbucket.org/johnsmith/applicationset.git`.
+    5. **Authentication:** select **Username and Token**. For the Token, you'll need to use a Personal Access Token (PAT) from Bitbucket. If you are logged into Bitbucket, just go to [HTTP access tokens](https://confluence.atlassian.com/bitbucketserver/http-access-tokens-939515499.html).
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md:376]: Use a Personal Access Token (PAT) from Bitbucket. If you are logged into Bitbucket, go to [HTTP access tokens](https://confluence.atlassian.com/bitbucketserver/http-access-tokens-939515499.html). You must provide an account-level app password or token. Repo-level tokens are not supported.

File: docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md [hunk 9]
```diff
@@ -372,10 +357,39 @@ Now we'll add a Github Connector to tell Harness where to pull the config.json f
...
+       You must provide an account-level app password or token. Repo-level tokens are not supported.
+   
+       You will store the PAT in a [Harness Text Secret](/docs/platform/Secrets/add-use-text-secrets). For details on Secrets Management, go to [Harness Secrets Management Overview](/docs/platform/Secrets/Secrets-Management/harness-secret-manager-overview).
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md:379]:
```suggestion
       Store the PAT in a [Harness Text Secret](/docs/platform/Secrets/add-use-text-secrets). For details on Secrets Management, go to [Harness Secrets Management Overview](/docs/platform/Secrets/Secrets-Management/harness-secret-manager-overview).
```

File: docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md [hunk 10]
```diff
@@ -372,10 +357,39 @@ Now we'll add a Github Connector to tell Harness where to pull the config.json f
...
+    
+    6. Select **Enable API access** and use the same Harness Secret.
+    
+    For more information, go to [Bitbucket connector settings reference](/docs/platform/Connectors/Code-Repositories/ref-source-repo-provider/bitbucket-connector-settings-reference).
+3. Select **Continue**.
+4. In **Connect to the provider**, select **Connect through Harness Platform**., and then click **Save and Continue**.
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md:385]:
```suggestion
4. In **Connect to the provider**, select **Connect through Harness Platform**, and then click **Save and Continue**.
```

File: docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md [hunk 11]
```diff
@@ -372,10 +357,39 @@ Now we'll add a Github Connector to tell Harness where to pull the config.json f
...
+5. When the **Connection Test** in complete, click **Continue**.
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/gitops/harness-git-ops-application-set-tutorial.md:386]:
```suggestion
5. After the Connection Test is complete, click **Continue**.
```

File: docs/continuous-delivery/gitops/sync-gitops-applications.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
+---
+title: Sync GitOps applications
+description: This topic describes how to sync GitOps applications.
+sidebar_position: 8
+---
+
+A sync brings the live state to its desired state by applying the declarative description. Sync will sync the desired Git state with the live cluster state.
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/gitops/sync-gitops-applications.md:7]: Suggestion:

Sync is a process that ensures that the live state of a system matches its desired state by applying a declarative description. This process involves synchronizing the desired Git state with the live cluster state.

File: docs/continuous-delivery/gitops/sync-gitops-applications.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+title: Sync GitOps applications
+description: This topic describes how to sync GitOps applications.
+sidebar_position: 8
+---
+
+A sync brings the live state to its desired state by applying the declarative description. Sync will sync the desired Git state with the live cluster state. 
+
+To sync applications from the **Applications** page: 
+
+1. In your GitOps project, go to **Deployments >** **GitOps>** **Applications**, and select your application.
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/gitops/sync-gitops-applications.md:11]:
```suggestion
1. In your GitOps project, go to **Deployments** > **GitOps** > **Applications**, and then select your application.
```

File: docs/continuous-delivery/gitops/sync-gitops-applications.md [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+   
+   ![](./static/sync-applications-3.png)
+
+2. To sync the selected application: 
+   * Select the more options icon, then select **Sync**.
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/gitops/sync-gitops-applications.md:16]:
```suggestion
   * Select the more options icon, and then select **Sync**.
```

File: docs/continuous-delivery/gitops/sync-gitops-applications.md [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+   
+     ![](./static/sync-applications-1.png)
+   * Select the application, then select **SYNC**.
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/gitops/sync-gitops-applications.md:19]:
```suggestion
   * Select the application, and then select **SYNC**. 
```

File: docs/continuous-delivery/gitops/sync-gitops-applications.md [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+   ![](./static/sync-applications-3.png)
+
+2. To sync the selected application: 
+   * Select the more options icon, then select **Sync**.
+   
+     ![](./static/sync-applications-1.png)
+   * Select the application, then select **SYNC**. 
+
+     ![](./static/sync-applications-2.png)
+3. Configure the sync options, then select **Synchronize**.
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/gitops/sync-gitops-applications.md:22]:
```suggestion
3. Configure the sync options, and then select **Synchronize**.
```

File: docs/continuous-delivery/gitops/sync-gitops-applications.md [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+   ![](./static/sync-applications-4.png)
+
+To sync applications using the **GitOpsSync** step: 
+
+1. Select a pipeline and go to the **Execution** tab of a deploy stage.
+   :::info
+
+   Make sure that the service, environment, and cluster selected in the pipeline matches the service, environment, and cluster in the application.
+
+   :::
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/gitops/sync-gitops-applications.md:33]:
```suggestion
   ::: 
   
```

File: docs/continuous-delivery/gitops/sync-gitops-applications.md [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+
+     ![](./static/sync-applications-2.png)
+3. Configure the sync options, then select **Synchronize**.
+   
+   ![](./static/sync-applications-4.png)
+
+To sync applications using the **GitOpsSync** step: 
+
+1. Select a pipeline and go to the **Execution** tab of a deploy stage.
+   :::info
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/gitops/sync-gitops-applications.md:29]:
```suggestion
   
   :::info
```

File: docs/continuous-delivery/gitops/sync-gitops-applications.md [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+
+   Make sure that the service, environment, and cluster selected in the pipeline matches the service, environment, and cluster in the application.
+
+   ::: 
+2. Select **Add Step**, then select the **GitOpsSync** step.
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/gitops/sync-gitops-applications.md:34]:
```suggestion
2. Select **Add Step**, and then select the **GitOpsSync** step.
```

File: docs/continuous-delivery/gitops/sync-gitops-applications.md [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+3. Select the GitOpsSync step to configure step parameters.
+4. In **Advanced Configuration**, select the application you want to sync and configure the sync options.
+   
+    ![](./static/gitopssync-step.png)    
+ 
+5. Select **Apply Changes**.
+
+After the sync, the resources will look like this in Harness:
```
Comment (SudheendraKatte) [inline docs/continuous-delivery/gitops/sync-gitops-applications.md:42]: Suggestion:

Here is how the resources would look in Harness after the sync process is complete.

**Review by SudheendraKatte — APPROVED** [review]
Approved with comments.

**Pull request merged**